examples/nuxt/pages/search.vue

```diff
@@ -64,6 +64,16 @@ import {
   createServerRootMixin,
 } from 'vue-instantsearch'; // eslint-disable-line import/no-unresolved
 import algoliasearch from 'algoliasearch/lite';
+import _renderToString from 'vue-server-renderer/basic';
+
+function renderToString(app) {
+  return new Promise((resolve, reject) => {
```

Comment:
I think we should accept both the callback form and the promise form (instanceof Promise to detect maybe) to avoid this glue code?

Reply:
Do you mean that we should promisify it inside createServerRootMixin in case it's vue 2?

Reply:
I think that's an option, but not sure. What do you think?

Reply:
```js
const result = renderToString(app);
if (result instanceof Promise) {
  ...
}     
```

but at that point, it's already invoked without a callback function. We need to know if the function is going to take a callback or just return a promise. So the only condition I can think of is to check vue version.

Handling callback vs promise based on vue version being used, it feels less adjacent (cannot think of the expression). what do you think?